hack/kep-lint.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Check that the required section headers (from the kep template) are
+# present in the kep.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TEMPLATE=/Users/arsh/Projects/Temp/kep-bash/kep-template.md
+
+KEP=/Users/arsh/Projects/Temp/kep-bash/keps/kep.md
```

Comment:
Where should we be getting these from?

Reply:
Taking inspiration from https://github.com/openshift/enhancements/blob/4412b51182ba6e007f1308a73e290e81fd66d092/hack/template-lint.sh, we should check any changed KEPs in the PR.
```bash
MODIFIED_KEPS=$(git log --name-only --pretty= "${PULL_BASE_SHA:-origin/master}.." -- \
                    | (grep '^keps.*\.md$' || true) \
                    | sort -u)
```

We should verify all these modified KEPs.

For the template, take `keps/NNNN-kep-template/README.md` as the source.